src/nvim/testdir/test_autocmd.vim

```diff
@@ -248,6 +248,24 @@ func Test_augroup_warning()
   au! VimEnter
 endfunc
 
+func Test_BufReadCmdHelp()
+  " This used to cause access to free memory
+  au BufReadCmd * e +h
+  help
+
+  helpclose
+  au! BufReadCmd
+endfunc
+
+func Test_BufReadCmdHelpJump()
+  " This used to cause access to free memory
+  au BufReadCmd * e +h{
```

Comment:
Why is this failing?

Reply:
Yes, even in vim 7.4/8.1 `:e +h{` gives `E32: No file name`.

Reply:
Caught exception in Test_BufReadCmdHelpJump(): Vim(help):E434: Can't find tag pattern @ function RunTheTest[35]..Test_BufReadCmdHelpJump, line 3

vim patch 8.01224 makes some changes including one which expects that failure: https://github.com/vim/vim/commit/cf1ba35fc2ebd41b9a7738bbd1f026f5311560aa